File: sdk/runanywhere-flutter/packages/runanywhere/lib/features/tts/models/tts_input.dart
import 'package:runanywhere/core/protocols/component/component_configuration.dart';
import 'package:runanywhere/features/tts/models/tts_options.dart';

/// Input for TTS synthesis
/// Matches iOS TTSInput from Features/TTS/Models/TTSInput.swift
class TTSInput implements ComponentInput {
  /// Text to synthesize (optional if SSML is provided)
  final String? text;

  /// Optional SSML markup (overrides text if provided)
  final String? ssml;

  /// Voice ID override
  final String? voiceId;

  /// Language override
  final String? language;

  /// Custom options override
  final TTSOptions? options;

  const TTSInput({
    this.text,
    this.ssml,
    this.voiceId,
    this.language,
    this.options,
  });

  /// Create input from plain text
  factory TTSInput.plainText(String text, {String? voiceId, String? language}) {
    return TTSInput(text: text, voiceId: voiceId, language: language);
  }

  /// Create input from SSML
  factory TTSInput.fromSSML(String ssml, {String? voiceId, String? language}) {
    return TTSInput(ssml: ssml, voiceId: voiceId, language: language);
  }

  @override
  void validate() {
    if ((text == null || text!.isEmpty) && (ssml == null || ssml!.isEmpty)) {
      throw ArgumentError('TTSInput must contain either text or SSML');
    }
  }
}

File: sdk/runanywhere-swift/Sources/RunAnywhere/Features/TTS/Models/TTSInput.swift
//
//  TTSInput.swift
//  RunAnywhere SDK
//
//  Input model for Text-to-Speech operations
//

import Foundation

/// Input for Text-to-Speech synthesis
///
/// Conforms to ComponentInput protocol for integration with the SDK's component system.
public struct TTSInput: ComponentInput, Sendable {

    // MARK: - Properties

    /// Text to synthesize
    public let text: String

    /// Optional SSML markup (overrides text if provided)
    public let ssml: String?

    /// Voice ID override
    public let voiceId: String?

    /// Language override
    public let language: String?

    /// Custom options override
    public let options: TTSOptions?

    // MARK: - Initialization

    public init(
        text: String,
        ssml: String? = nil,
        voiceId: String? = nil,
        language: String? = nil,
        options: TTSOptions? = nil
    ) {
        self.text = text
        self.ssml = ssml
        self.voiceId = voiceId
        self.language = language
        self.options = options
    }

    // MARK: - ComponentInput

    public func validate() throws {
        if text.isEmpty && ssml == nil {
            throw SDKError.tts(.emptyInput, "Text cannot be empty")
        }
    }

    // MARK: - Factory Methods

    /// Create input from plain text
    public static func text(_ text: String) -> TTSInput {
        TTSInput(text: text)
    }

    /// Create input from SSML markup
    public static func ssml(_ ssml: String) -> TTSInput {
        TTSInput(text: "", ssml: ssml)
    }

    /// Create input with voice override
    public static func text(_ text: String, voice: String) -> TTSInput {
        TTSInput(text: text, voiceId: voice)
    }

    /// Create input with language override
    public static func text(_ text: String, language: String) -> TTSInput {
        TTSInput(text: text, language: language)
    }
}
